app-i18n/mozc/mozc-2.23.2785.102.1.recipe

```diff
@@ -1,50 +1,51 @@
 SUMMARY="Japanese input method editor"
 DESCRIPTION="A Japanese Input Method Editor (IME) designed for multi-platform. \
 This is open source version."
-HOMEPAGE="https://github.com/google/mozc/"
-COPYRIGHT="2010-2016, Google Inc."
+HOMEPAGE="https://github.com/google/mozc"
+COPYRIGHT="2010-2018 Google Inc."
 LICENSE="BSD (3-clause)"
-REVISION="2"
-fileVersion="v2.20.2677.102.6"
-SOURCE_URI="https://github.com/hanya/mozc/releases/download/$fileVersion/mozc-$fileVersion.tar.bz2"
-CHECKSUM_SHA256="cdb43a88d82db185299a8e396fccfe76ddbbe5a3c6b219220a36b55115cff4ac"
+REVISION="1"
+SOURCE_URI="https://github.com/hanya/mozc/releases/download/v$portVersion/mozc-v$portVersion.tar.bz2"
+CHECKSUM_SHA256="17ce4ee5d6a55678d804890d7e2fa7ffd7f852a9673e3b79671613c9167469f5"
 SOURCE_DIR="mozc"
-if [ $effectiveTargetArchitecture == "x86" ]; then
+if [ "$effectiveTargetArchitecture" = x86 ]; then
 	PATCHES="mozc-$portVersion-x86.patch"
 fi
-SOURCE_URI_2="https://jaist.dl.sourceforge.net/project/zinnia/zinnia-tomoe/0.6.0-20080911/zinnia-tomoe-0.6.0-20080911.tar.bz2"
+zinniaTomoeVersion="0.6.0-20080911"
+SOURCE_URI_2="https://downloads.sourceforge.net/zinnia/zinnia-tomoe-$zinniaTomoeVersion.tar.bz2"
 CHECKSUM_SHA256_2="fd1a264b8aeea0ebec59e8c8765864a9543a75fb2dba349745bb3a3587790927"
 
 ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="!x86"
 
 PROVIDES="
 	mozc
```

Comment:
Thanks @hanya for the update :)

There were 2 small mistakes in my previous review:
* I should have suggested a `"?x86"` (instead of `"x86"`) for `SECONDARY_ARCHITECTURES`.
* I forgot to write that `addon:mozc` needs to have a `$secondaryArchSuffix` appended to it, but you fixed that by yourself :) 👍

So, I think that with the following changes the recipe will be OK (and probably acceptable):
```diff
 ARCHITECTURES="!x86_gcc2 x86 x86_64"
-SECONDARY_ARCHITECTURES="!x86"
+SECONDARY_ARCHITECTURES="?x86"
 
 PROVIDES="
- 	mozc
+	mozc$secondaryArchSuffix
 	addon:mozc$secondaryArchSuffix
 	"
```

Remarks:
1) The reason why we need `mozc$secondaryArchSuffix` in `PROVIDES` is that, on x86_gcc2(h), we need to make it clear that the package name is mozc_x86, not mozc. This way, a `pkgman search mozc` will yeld `mozc_x86` on x86_gcc2h, and `mozc` on x86 and x86_64.

2) I did not try yet to build mozc on x86 primary arch. But I guess you did, otherwise you would not have written a patch that is only applied on x86.
So, if you have mozc building and working fine on x86 primary arch, this means it might be possible to also build it on x86 secondary arch. This is why I suggested having a `?x86` (instead of `!x86`) in `SECONDARY_ARCHITECTURES`.
I did not investigate yet why mozc_x86 is not building, but I noticed that unlike python3, which has support for secondary architectures, python2 does not. So maybe what is missing, to build mozc on x86 secondary arch, is a python2_x86 (and replacing `cmd:python2` by `cmd:python2$secondaryArchSuffix`).

**_To everyone:_**
**Does anyone object in having a `?x86` (instead of `!x86`) in `SECONDARY_ARCHITECTURES` ?**

Reply:
I tried to build mozc 2.23.2785.102.1 (from this PR) on x86 primary arch, but it is failing for me:
```
INFO: Running: ninja -C out_linux/Release mozc_server mozc_emacs_helper mozc mozc_tool zinnia_model
ninja: Entering directory `out_linux/Release'
[4/767] RULE(host) Generating C++ code from config.proto
FAILED: cd ../../protocol; python ../build_tools/protoc_wrapper.py "--project_root=.." "--protoc_command=protoc" "--proto=config.proto" "--cpp_out=../out_linux/Release/gen/proto_out"
protocol/config.proto:66:3: Expected "required", "optional", or "repeated".
protocol/config.proto:66:12: Expected field name.
...
protocol/config.proto:344:3: Expected "required", "optional", or "repeated".
protocol/config.proto:344:12: Expected field name.
[4/767] CXX obj/prediction/gen/proto_o...n_protocol.user_history_predictor.pb.o
ninja: build stopped: subcommand failed.
Traceback (most recent call last):
  File "build_mozc.py", line 1241, in <module>
    main()
  File "build_mozc.py", line 1228, in main
    BuildMain(cmd_opts, cmd_args)
  File "build_mozc.py", line 859, in BuildMain
    BuildWithNinja(options, targets)
  File "build_mozc.py", line 833, in BuildWithNinja
    RunOrDie([ninja, '-C', build_arg] + ninja_targets)
  File "/sources/mozc/src/build_tools/util.py", line 99, in RunOrDie
    '==========']))
build_tools.util.RunOrDieError: 
==========
 ERROR: ninja -C out_linux/Release mozc_server mozc_emacs_helper mozc mozc_tool zinnia_model
```

(I get the same errors if I try to build `mozc-2.20.2677.102.6` using the recipe that is already in the repo, `mozc-2.20.2677.102.recipe`.)

Did you succeed in building mozc on x86 primary arch?
(Right now, the only arch on which I have been able to build mozc is x86_64.)
Thanks!

Reply:
I've uploaded packages I could built here: https://github.com/hanya/mozc/releases
protobuf version is 3.3.1-1 on x86.

Reply:
Thanks for the upload. I grabbed the hpkg for x86 primary arch and was even able to install it. But for some (unknown) reason it is not working for me.

Can you replace `mozc` by `mozc$secondaryArchSuffix` in `PROVIDES` ?

I've tried to kick another build of mozc for x86 secondary arch (just in case some external factor could have been the cause of my previous build failures), but I'm still hitting the same errors:
```
looper.cc:(.text+0x221f): undefined reference to `BInputServerMethod::SetMenu(BMenu const*, BMessenger)'
collect2: error: ld returned 1 exit status
[386/767] LINK mozc_no_resource
```

So, after all, let's keep the `!x86` you have put in `SECONDARY_ARCHITECTURES`.